packages/yoga/src/Result/native/Result/TextWithBadge.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useCallback, useState } from 'react';
```

Comment:
We have an initiative to add Typescript to our project, so we are asking for new components to be in TS :)

Furthermore, we have the sonar configured for our test coverage analysis, what do you think about having a test for this new component?

Reply:
Nice suggestion, Flavia. I'm on it.

Reply:
Done! :)